FindCAres.cmake

```diff
@@ -12,8 +12,10 @@ if (NOT HAVE_CARES)
         find_path(CARES_ROOT_DIR NAMES "include/ares.h")
 
         # Prefer linking statically but look for a shared library version too.
-        find_library(CARES_LIBRARIES NAMES libcares_static.a libcares.so
-                     HINTS ${CARES_ROOT_DIR}/lib)
+        find_library(
+            CARES_LIBRARIES NAMES "libcares_static${CMAKE_STATIC_LIBRARY_SUFFIX}"
+                                  "libcares${CMAKE_SHARED_LIBRARY_SUFFIX}"
```

Comment:
As in DM, may just not add the suffixes here as CMake will do this itself:

> Each library name given to the NAMES option is first considered as a library file name and then considered with platform-specific prefixes (e.g. lib) and suffixes (e.g. .so).

Reply:
I haven't check the CMake sources, but couldn't that break our current "prefer static link if possible" behavior? If we are fine with deviating from that, let's at least make sure that a static build uses static libs if they are present here.

Reply:
My only concern is whether the order is platform-specific. I did a build on ubuntu22 with the suggested change and it picked up the static library for both, even though it built both the static and shared libraries for libkqueue. I checked the value of `CMAKE_FIND_LIBRARY_SUFFIXES`, which should set the order preference and it's set to `.so;.a`.